Fix imports collisions

alshakero opened the pull request:
The commit history isn't the most beautiful (read: terrible) 

Most interesting file to review is this: https://github.com/Starcounter/StarcounterClientFiles/pull/52/files#diff-148ced327f109eb0b0e22fa3c6a24f9d

What I am doing now:

1. If `polymer-source` doesn't exist, backup whole `polymer` directory into it. If it does copy it over to `polymer` directory, to process a fresh copy of `polymer` directory everytime.
2. Vulcanize only `polymer.html` (without `polymer-element.html`).
3. Collect the list of vulcanized files and append `polymer-element.html` to it.
4. Iterate the list of each vulcanized file and: 
	1. Determine the relative path from each file to `polymer.html` (say POLYMER_PATH).
	2. Overwrite the file's contents with `<import href="POLYMER_PATH">` (A genius idea from @warpech).
5. Log "Vulcanizing done".

alshakero committed:
Update vulcanizing script to remove all vulcanized code from source files

alshakero committed:
Backup Polymer source before vulcanizing

alshakero committed:
Enhance backup logic

warpech committed:
a bunch of improvements to the vulcanize script

1. change .bowerrc and bower.json to install Polymer in "polymer-source"
2. vulcanize.node.js was  using an old version of the `vulcanize` tool. I have switched to use the current version called `polymer-bundler`, installed from `package.json`
3. instead of spawning vulcanization, I use PolymerBundler programatically
4. instead of copying all files from "polymer-source" to "polymer", I just copy the html files
5. all the shortened files had a typo, they used <import> instead of <link rel="import">
6. restore original versions of two files in ShadyCSS. I hope they never get imported directly because this might cause errors
7. change the build step in csproj to use npm run
8. add build step in CONTRIBUTING.md

warpech commented:
I made a bunch improvements that I could think of. They are explained in the commit message.

I tested `vaadin-date-picker` locally with KitchenSink and it worked well!

alshakero committed:
Enhance vulcanizing script

1. Use Node's fs instead of fs-extra (more trust-worthy)
2. Fail gracefully if something goes wrong (given we do disk I/O)
3. Use doc.files instead of _bundleUrlForFiles.keys()
4. Log when done; it's the only way to tell in VS log and TeamCitng
5. Use await for better readiblity
6. Use for..of instead of forEach (to save creating a callback
everytime)

alshakero commented:
> Another idea improvement for you @alshakero: polymer-bundler does not minimize the vulcanized file. If you replace it with npmjs.com/package/polymer-build (which is a superset toolkit), you might be able to generate a minimized file.

Should this step be in this PR? I played around with it and it's failing. I'll need to understand Node streams a bit to do it.

alshakero committed:
Merge branch '3.x' into Fix-imports-collisions

warpech reviewed (approved):
Lgtm. It will be released as 3.1.0
